experimental/single_number_formatter/src/ule.rs

```diff
@@ -0,0 +1,130 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerovec::{
+    maps::ZeroMapKV,
+    ule::{AsULE, ZeroVecError, ULE},
+};
+
+use crate::provider::{CurrencyPatterns, PatternSelection};
+
+/// `CurrencyPatternsULE` is a type optimized for efficient storing and
+/// deserialization of `CurrencyPatterns` using the `ZeroVec` model.
+///
+/// The serialization model packages the pattern item in three bytes.
+///
+/// The first two bits is used to determine the short_pattern_standard. If the bits are `00`, then, the value will be `Standard`.
+/// If the bits are `01`, then, the value will be `StandardAlphaNextToNumber`.
```

Comment:
How about `10` and `11`? Are they invalid, reserved, or do they map to `Standard`? (I suggest they map to `Standard`)